src/DatabaseMinifier.php

```diff
@@ -519,11 +506,14 @@ protected function copyRecordsByCriteriaInternal(
                 'referenced_by' => [],
             ];
 
-            //TODO: what if we have not existed references
-            $result['references'] = $this->copyReferences($tableName, $row);
-
             if (!$this->checkIfRowCopied($tableName, $row)) {
-                $this->pasteRow($tableName, $row);
+                //TODO: what if we have not existed references
+                $result['references'] = $this->copyReferences($tableName, $row);
+
+                if (!$this->checkIfRowCopied($tableName, $row)) {
```

Comment:
Do we  really need this check? it weird if we will copy row within copying references :D